tests/test_utils_misc/test_return_with_argument_inside_generator.py

```diff
@@ -3,6 +3,28 @@
 from scrapy.utils.misc import is_generator_with_return_value
 
 
+def top_level_function_with():
+    """
+docstring
+    """
+    url = """
+https://example.org
+"""
+    yield 1
+    return 2
+
+
+def top_level_function_without():
+    """
+docstring
+    """
+    url = """
+https://example.org
+"""
+    yield 1
+    return
```

Comment:
My suggestion from https://github.com/scrapy/scrapy/issues/4477#issuecomment-611018866 was not working correctly for these cases.